README.adoc

```diff
@@ -75,88 +61,156 @@ It could be omitted as the title already says that it is an experimental feature
 * Avoid the phrase “It is recommended to” when proposing a solution.
 * Capitalize all Cypher keywords, i.e., `MATCH` rather than `match` or `Match`.
 
-=== Code
 
-All Neo4j codes follow the syntax `Neo.[Class].[Subclass].[Name]`.
-The notification codes’ [Class] is `ClientNotification`.
-Therefore, they all start with `Neo.ClientNotification`.
-The [Subclass] is one of `Statement`, `Procedure`, `Schema`, `Database`, `Security`, or `Request`.
-Code’s [Name] should be specific and explain what this code refers to.
+=== Status code elements
 
-✅Examples:
-Code: Neo.ClientNotification.Statement.RepeatedRelationshipVariableInPattern
+Each status code comprises the following components, some of which are documented only here in this repo.
 
-⛔Examples:
-Code: Neo.ClientNotification.Statement.RepeatedRelationshipReference
+[options="header",cols="h,2,2",]
+|===
+| Element
+| Notification
+| Error
 
-=== Title
+| Status code (`Neo.[Class].[Subclass].[Name]`)
+| `Neo.ClientNotification.Statement.CartesianProduct`.
+| `Neo.DatabaseError.Statement.RemoteExecutionFailed`.
 
-The title should be brief (one short sentence), scannable, and inform on the current situation and what code of the query triggered the notification (add it within backticks to show that it’s a code snippet).
+| Title/Description of the status code
+| `This query builds a cartesian product between disconnected patterns.`
+| `The database was unable to execute a remote part of the statement.`
 
-* Do not use exclamation marks.
-* Use sentence case (you capitalize the first letter, proper nouns, and names, just like you would in a normal sentence) and punctuation.
-* Avoid explaining:
-** the consequences of the returned code.
-** what causes the code to be returned.
-** possible solutions.
+| SeverityLevel
+| One of `INFORMATION`\|`WARNING`
+m| ERROR
 
-⛔Example (contains a possible solution):
-      The provided pattern is unbounded, consider adding an upper limit to the number of node hops.
+| Category
+| One of `DEPRECATION`\|`HINT`\|`UNSUPPORTED`\|`UNRECOGNIZED`\|`SECURITY`\|`GENERIC`\|`PERFORMANCE`\|`TOPOLOGY`
+| N/A
 
-✅Example:
-     The pattern `<pattern>` is unbounded.
+| Description/Message
+2+| Contains the main information, such as the problem, cause, consequences, a simple solution, or a reference to the Status Codes documentation.
 
-⛔Example:
-This feature is deprecated and will be removed in future versions.
+| Example (only in Status Codes documentation)
+2+| Contains one or more example queries to illustrate the possible scenarios when this notification code would be returned.
```

Comment:
Currently, we do not have any examples of errors or error messages in here, do we plan on that and to what extent? I think we will start to just update error messages in the spreadsheet following the new guidelines and not add each error message as an example here? Maybe we need to do it eventually when we have all GQL codes.

Depending on the above, 'notification' should maybe be updated to whatever common word for notifications and errors we choose to use